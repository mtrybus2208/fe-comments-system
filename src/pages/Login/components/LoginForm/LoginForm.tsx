import React from 'react';
import { Formik, Field, Form, FormikHelpers, FormikProps } from 'formik';

import FormWrapper from '../../../../components/FormWrapper/FormWrapper';
import FieldInput from '../../../../shared/components/FieldInput/FieldInput';
import FieldLabel from '../../../../shared/components/FieldLabel/FieldLabel';
import Button from '../../../../shared/components/Buttons/Button/Button';
import { LoginFormValues } from '../../Login.types';

export interface LoginFormProps {
  onSubmit: (
    values: LoginFormValues,
    actions: FormikHelpers<LoginFormValues>,
  ) => void;
}

const LoginForm: React.FunctionComponent<LoginFormProps> = ({ onSubmit }) => {
  const handleSubmit = (
    values: LoginFormValues,
    actions: FormikHelpers<LoginFormValues>,
  ): void => {
    actions.setSubmitting(false);
    onSubmit(values, actions);
  };

  return (
    <Formik
      initialValues={{ email: '', password: '' }}
      onSubmit={handleSubmit}
      render={(props: FormikProps<LoginFormValues>) => (
        <FormWrapper name="Login">
          <Form>
            <FieldLabel htmlFor="email" name="Email Address" />
            <Field
              type="text"
              name="email"
              placeholder="Email"
              component={FieldInput}
            />
            <FieldLabel htmlFor="password" name="Password" />
            <Field
              name="password"
              type="password"
              placeholder="Your password"
              component={FieldInput}
            />
            <Button type="submit">Submit</Button>
          </Form>
        </FormWrapper>
      )}
    />
  );
};

export default LoginForm;
