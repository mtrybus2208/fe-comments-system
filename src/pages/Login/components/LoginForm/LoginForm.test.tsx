import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import LoginForm from './LoginForm';

const renderer = new ShallowRenderer();

describe('LoginForm', () => {
  it('should render layout', () => {
    const props = {
      onSubmit: jest.fn(),
    };

    const wrapper = renderer.render(<LoginForm {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
