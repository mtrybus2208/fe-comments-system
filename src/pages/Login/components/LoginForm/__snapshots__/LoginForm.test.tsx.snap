// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should render layout 1`] = `
<React.Fragment>
  <Formik
    initialValues={
      Object {
        "email": "",
        "password": "",
      }
    }
    onSubmit={[Function]}
    validationSchema={
      ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": undefined,
        "_nodes": Array [
          "password",
          "email",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "email": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "matches": false,
              "required": true,
              "trim": false,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
              [Function],
            ],
          },
          "password": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
              "trim": false,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
              [Function],
            ],
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
      }
    }
  >
    [Function]
  </Formik>
</React.Fragment>
`;
